tequila_fab/ansible.py

```diff
@@ -0,0 +1,128 @@
+    """
+    Return path to the ansible config file that ansible would use,
+    or None if no config file found.
+
+    Ansible uses the first config file it finds on this list:
+
+    ANSIBLE_CONFIG (an environment variable)
+    ansible.cfg (in the current directory)
+    .ansible.cfg (in the home directory)
+    /etc/ansible/ansible.cfg
+    """
+    possible_paths = [
+        os.environ.get('ANSIBLE_CONFIG', False),
+        'ansible.cfg',
+        os.path.join(os.environ['HOME'], '.ansible.cfg'),
+        '/etc/ansible/ansible.cfg',
+    ]
+    for path in possible_paths:
+        if path and os.path.exists(path):
+            return os.path.abspath(path)
+    return None
+
+
+@functools.lru_cache(maxsize=1)
+def get_ansible_configuration() -> configparser.ConfigParser:
+    config = configparser.ConfigParser()
+    config['DEFAULTS'] = {
+        'roles_path': os.path.join(os.environ['HOME'], '/.ansible/roles') + ':/usr/share/ansible/roles:/etc/ansible/roles'
+    }
+
+    path = find_ansible_config_file()
+    if path is not None:
+        config.read(path)
+    return config
+
+
+@functools.lru_cache(maxsize=1)
+def get_roles_path() -> List[str]:
+    """
+    Return list of directories where Ansible will look for roles
+    """
+    return get_ansible_configuration()['defaults']['roles_path'].split(':')
+
+
+@task
+def install_roles():
+    """
+    Usage: fab install_roles
+    """
+    local('ansible-galaxy install -i -r deployment/requirements.yml')
+
+
+def find_install_role(rolename) -> Optional[str]:
+    """
+    Returns path of directory where role is installed,
+    or None.
+    """
+    for path in get_roles_path():
+        dir = os.path.join(path, rolename)
+        if os.path.isdir(dir):
+            return dir
+
+
+def req_name(req) -> str:
+    """
+    Return name of the role in the given roles requirements entry
+    """
+    return req.get('name', req['src'])
+
+
+@task
+def check_role_versions():
+    """
+    Usage: fab check_role_versions
+
+    Fails if the exact versions of roles specified in deployment/requirements.yml
+    are not installed.
+    """
+
+    okay = True  # False if we've spotted any problems
+    bad = []  # Paths to where missing roles should be installed, or where bad version roles are installed
+    requirements = yaml.load(open("deployment/requirements.yml"))
+    requirements = sorted(requirements, key=req_name)
+    for req in requirements:
+        name = req_name(req)
+        install_dir = find_install_role(name)
+        if not install_dir:
+            okay = False
+            print("ERROR: role %s not installed" % (name,))
+            continue
+        meta_path = os.path.join(install_dir, 'meta/.galaxy_install_info')
+        if os.path.exists(meta_path):
+            meta = yaml.load(open(meta_path))
+            if meta['version'] != req['version']:
+                print("ERROR: role %s at %s is version %s, should be version %s" % (
```

Comment:
Small nit: Given that you can easily have 8-10 galaxy requirements, visually parsing an error in the output could be easier if these errors were a different color using fabric's colorized output. For example, this could be **red** and the SKIPs could be **yellow**.